metrics-api/src/main/java/org/killbill/commons/metrics/api/Gauge.java

```diff
@@ -17,12 +17,21 @@
 
 package org.killbill.commons.metrics.api;
 
-public interface Gauge<T> extends Metric {
-
-    /**
-     * Returns the metric's current value.
-     *
-     * @return the metric's current value
-     */
-    T getValue();
```

Comment:
NACK -- or maybe I'm misunderstanding how it would be used.

Unlike other metrics, Gauges are usually created once and callers don't keep a reference to it. Instead, the registry polls the value. Check `org.killbill.billing.util.metrics.JmxAttributeGauge` for instance.

Reply:
The dropwizard model here seems a bit odd, that you have to pass in a Gauge (which is actually a lambda or an anonymous class) to create a Gauge.

My suggestion would be to create something you can manipulate the value (set, inc, dec, etc), or create providing a functional interface (in this case a `DoulbeSupplier`).

Reply:
Agreed, the model is quite odd today. A `Gauge` currently doesn't _have_ a value (the value is retrieved realtime when querying the Gauge).

That being said, **requiring** the caller to _manipulate the value_ (e.g. through _set_) would make it difficult to use in practice. For example, for the `JmxAttributeGauge`, we would need a backend threadpool to `set` the Gauge periodically.

Providing an alternative option (e.g. `DoubleSupplier`) seems necessary.

Reply:
I agree. Would the method for doing so in [MetricRegistry](https://github.com/killbill/killbill-commons/pull/131/files#diff-d5bb2d15c63f242008c51ec78d4641bba62556ecfb15e65f4a9f50e8915a1a1eR43) be enough?

Reply:
Oh, I missed this! Yup, I think so, to be prototyped 😄